pkg/config/config.go

```diff
@@ -665,6 +665,15 @@ func load(config Config, origin string, loadSecret bool) error {
 		}
 	}
 
+	// If we run Agent > 6, we want to use Python 3.
+	if av, err := version.Agent(); err != nil {
```

Comment:
I believe this is the first piece of logic that actually relies on the built-in Agent version. Ideally it'd be nice to not have the code rely on the version of the Agent at all to determine what python to use because we'd like all this logic to be applied at build time.

Typically, here this could mean adding a package-level variable called `ForceDefaultPython` that would be set at compile time, and would define whether the Agent allows configuring the version of python to run. (similar to `DefaultPython`)
The build flag would be a bit redundant with the build flag that sets the version, but I like the fact that it surfaces this logic at compile time instead of run time. What do you think?

Reply:
I agree because it felt weird to write a runtime test to force the python version. I'll look into the solution you've mentioned.